libcnb-test/src/lib.rs

```diff
@@ -0,0 +1,260 @@
+// Enable rustc and Clippy lints that are disabled by default.
+// https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#unused-crate-dependencies
+#![warn(unused_crate_dependencies)]
+// https://rust-lang.github.io/rust-clippy/stable/index.html
+#![warn(clippy::pedantic)]
+// Most of libcnb's public API returns user-provided errors, making error docs redundant.
+#![allow(clippy::missing_errors_doc)]
+// This lint is too noisy and enforces a style that reduces readability in many cases.
+#![allow(clippy::module_name_repetitions)]
+// This lint triggers when both layer_dir and layers_dir are present which are quite common.
+#![allow(clippy::similar_names)]
+#![allow(clippy::default_trait_access)]
```

Comment:
I don't like this lint at all. Especially when constructing large structs, I find it much nicer to do `..Default::default()` instead of `..StructUnderConstruction::default()` and it's often seen in examples or docs of other crates - Bollard is one example.

However, ☝🏻 is a little hypocritical - I usually say that I bow before default linter rules and just deal with them regardless of my personal taste.

Reply:
So I agree the benefit is more borderline when using it in the context of `..Default::default()`.

However I do think it's beneficial in other situations, such as:
https://github.com/Malax/libcnb.rs/blob/39cb9879707e16323e44ff5e14408ba43d2b126c/libcnb/src/layer/test.rs#L843

...where `LayerResultBuilder::new(Default::default())` would be much less clear.

Or:
https://github.com/Malax/libcnb.rs/blob/39cb9879707e16323e44ff5e14408ba43d2b126c/libcnb/src/layer/test.rs#L789

...where `types: Some(Default::default()),` would be less clear.

It also feels a bit strange to just disable this lint in one crate, and not all of them.

Reply:
I re-enabled the lint in 435cb44, it's really not that bad to type the name out and having less exceptions which lints are enabled is better IMO.